suave/sol/standard_peekers/example.sol

```diff
@@ -44,6 +44,44 @@ contract ExampleEthCallSource {
 
         return abi.encodeWithSelector(this.emptyCallback.selector);
     }
+
+    event LogCallback(bytes data);
+
+    function emitLogCallback(uint256 num) public {
+        // From the msg.input, the 'confidential context' sequence
+        // starts at index 37 (4 signbature bytes + 32 bytes for uint256 + 1 byte for the magic sequence)
+        uint256 magicSequenceIndex = 37;
+
+        bytes memory inputData = msg.data;
+        uint256 dataLength = inputData.length - magicSequenceIndex;
+
+        // Initialize memory for the data to decode
+        bytes memory dataToDecode = new bytes(dataLength);
+
+        // Copy the data to decode into the memory array
+        for (uint256 i = 0; i < dataLength; i++) {
+            dataToDecode[i] = inputData[magicSequenceIndex + i];
+        }
+
+        emit LogCallback(dataToDecode);
```

Comment:
I wonder
The most obvious way to store the events would be encoded as an array of (event) bytes
This code points to that not being the case - rather we have a single encoded value (`ExecResult` which uses `tuple(tuple(address addr, bytes32[] topics, bytes data)[] logs)` underneath)
Could you simply encode that tuple as a struct and decode it here in Solidity?
What you would get is abi-encoded list of logs which you can then emit as-is using `log3` (or something like that I think?)

Also, what if instead you kept the encoded event instead of logs, and simply emitted the encoded logs? I think that would actually make the logs trivially parseable, whereas as is you have to use a roundabout way to decode the actual logs emitted (I don't think you show how to do that yet)

I'd look like so:
```solidity
bytes[] memory raw_logs = abi.decode(dataToDecode, (bytes[]));
for raw_log in raw_logs { emit raw_log; }
```
with a simple change in how you encode the execution result (or the log).

Reply:
I think actually emitting the logs rather than a single log would be interesting - those could then be parsed as one would parse any other logs, rather than having to go through a layer of indirection

using log<N> I think could be simple enough, since they only go until 4 right?